doc/services/zbus/index.rst

```diff
@@ -0,0 +1,356 @@
+
+Another important aspect of using zbus is the reuse of system modules. If a module, code portion with a set of well-defined behaviors, only uses zbus channels and not hardware interfaces, it can easily be reused in other solutions. For that, the new solution must implement the interfaces (set of channels) the module needs to work. That indicates zbus could improve the module reuse.
+
+The last important note is the zbus solution reach. We can count on many different ways of using zbus to enable the developer to be as free as possible to create what they need with it. Messages can be dynamic or static allocated, notifications can be synchronous or asynchronous, the developer can control the channel in so many different ways claiming the channel, developers can add their metadata information to a channel by using the user-data field, the discretionary use of a validator enables the systems to be accurate over message format, and so on. Those characteristics increase the solutions that can be done with zbus and make it a good fit as an open-source community tool.
+
+Limitations
+===========
+
+Based on the fact that developers can use zbus to solve many different problems, some challenges arise. Zbus will not solve every problem, so it is necessary to analyze the situation to be sure zbus is applicable. For instance, based on the zbus benchmark, it would not be well suited to a high-speed stream of bytes between threads. The `Pipe` kernel object solves this kind of need.
+
+Delivery guarantees
+-------------------
+
+In zbus, there are no message delivery guarantees. However, it is possible to increase the delivery rate if some conditions happen:
+No user's thread must have a priority higher than the zbus event dispatcher.
+Synchronous observers must have non-blocking and quick-as-possible implementations.
+In asynchronous observer scenarios, producers must have a lower priority than consumers.
+
+Message delivery sequence
+-------------------------
+
+The listeners (synchronous observers) will follow the channel definition sequence as the notification and message consumption sequence. However, the subscribers, as they have an asynchronous nature, all will receive the notification as the channel definition sequence but only will consume the data when they assume the CPU again, so the delivery respects the order, but the reaction of it will be oriented by the priority assigned to the subscribers.
+
+Implementation
+**************
+
+Zbus operation depends on channels and observers. Therefore, it is necessary to determine its message and observers list during the channel definition. A message is a regular C struct; the observer can be a subscriber (asynchronous) or a listener (synchronous). Channels can have a combination of the following characteristics:
+
+* ``Validator function`` makes a channel accepts only valid messages;
```

Comment:
Not quite for the comment but: haven't seen any samples that tests the validator, would it make sense to add a validator function to one of those?